src/ripple/app/misc/Transaction.h

```diff
@@ -52,6 +53,8 @@ enum TransStatus {
     INCOMPLETE = 8   // needs more signatures
 };
 
+enum class SearchedAll { no, yes, unknown };
+
```

Comment:
Now that this lives in a header file, I think this needs to be renamed or put into another namespace. `SearchedAll` is too generic a name to live in the main rippled namespace. Maybe `TxSearchedAll`?

Reply:
How about `boost::tribool`?

Or this:

```cpp
enum class TxSearched { all, some, unknown };
```

Reply:
The `all, some, unknown` renaming is good. I like that. I don't think `tribool` is a good fit for this tho.

Reply:
I kind've think we should just remove the `unknown` option here. I think `Transaction::load` should return `std::variant<std::monostate, std::pair<std::shared_ptr<Transaction>, std::shared_ptr<TxMeta>>, TxSearchedAll>`. `TxSearchedAll` could change to a bool then, though I think it's easier to understand what the meaning of the returned value is when the enum is used (as opposed to just a bool). 

There's really no need for `TxSearchedAll::unknown`; it just means the transaction wasn't found, and we didn't check the range of ledgers (because the caller didn't ask to check the range). The caller knows that they didn't ask for the range of ledgers to be checked, so `TxSearchedAll::unknown` doesn't actually provide any additional info to the caller. I think simply returning an empty variant (via `std::monostate`) is the most natural thing here. Or just wrapping the whole return type in an `std::optional`.

Another option would be to break this up into two separate functions. Right now `Transaction::load` is doing two things: looking for the transaction, and also reporting info about the range of ledgers in the database, in the event that the transaction wasn't found. We wouldn't have to hem and haw about return types if this was broken up into two separate functions.

Reply:
@cjcobb23 Just commenting on the `monostate` solution. I think I prefer the enum over that. The nice thing about the enum is there's one place to look when we don't get the result we want. With the `optional` or the `monostate` there's two places: is the `std::monostate` the active part of the `variant` - that means one kind of "we didn't get what we want' or if the `bool` part of the `variant` is active - that means another kind of "we didn't get what we want". I think it's cleaner to put that information in one place.

Reply:
@seelabs that's a fair point about having all of the "not found" results as the same element of the variant.

I think ultimately I would prefer to split this into two functions. One to get the transaction, and one to check the ledger range. Both would be called from the handler, and the range check would only be called if the client asked for that information (and the transaction wasn't found).

Reply:
@cjcobb23 Splitting into two functions sounds attractive, I like that. We'd have to look to make sure that ledger range can't change between the not finding the transaction and checking the range we'd give faulty information (or make sure we hold the appropriate locks).

While I like the idea, what do you think about punting on it for now and tackling it in another PR? I'd like to get this one in and @natenichols's solution is pretty good as-is.

Reply:
@seelabs @natenichols I'm good with the code as-is as well